src/main/java/com/mael/payroll/Employee.java

```diff
@@ -60,7 +60,7 @@ public PaymentMethod getPaymentMethod() {
     }
 
     public void getPaid(Paycheck paycheck) {
-        double netPay = getPaymentType().calculatePay();
+        double netPay = getPaymentType().calculatePay(paycheck.getDate());
```

Comment:
I think it would be enough to just pass the date, since that's all what the methods need to calculate the amount to be paid. I think the paycheck is there because they use it in the book for the payday transaction.

Anyway, this method is gonna change in the future when I add the affiliations, since they are deductions to be subtracted from the net pay. In any case, I think the calculations should probably be made in the paycheck, by calling the relevant methods on the relevant classes.

I still don't know! :smile: I have to think about it.